Replace rpmfusion Obsoletes with Conflicts, add auto-fallback to nouveau systemd service

jwrdegoede opened the pull request.

scaronni commented:
I think this is not enough. I'm upgrading from RPMFusion, elrepo and Nvidia CUDA repositpories, this is only a bit of it.

For example, one of the latest bugs opened on Fedora's bugzilla (don't have the ticket number at hand, sorry, it's the one where Leigh helped) where the user had left overs from both RPMFusion and CUDA repositories would have not happened with the Obsoletes/Provides in place.

I guess I will just remove all Obsoletes/Provides and put everything into Conflicts.

Would it be better if I just leave the base driver and modules/tools in the repository? Being an interim solution I can leave CUDA and all the rest in the other repository (multimedia). Most of the universities use those.

scaronni commented on nvidia-driver.spec:
```diff
@@ -4,15 +4,15 @@
-%global _dracutopts     nouveau.modeset=0 rdblacklist=nouveau
-%global _dracutopts_rm  nomodeset vga=normal
+%global _dracutopts     rdblacklist=nouveau
+%global _dracutopts_rm  nomodeset vga=normal nouveau.modeset=0
```
Further down in the SPEC file I just check for the whole string of options, and the other two have been there for quite a while, so I guess all users will not have those and the `nouveau.modeset=0` parameter would not get removed.

I guess I need to remove the old Dracut options and just leave that, or iterate on each of one in the SPEC file.

scaronni commented:
Looks fine for me, as written in the email I would have actually preferred to have the kernel locked into place until all modules are available.

Someone who installs the Nvidia drivers would definitely give priority to the drivers being available than updating the kernel. I think users will be pretty vocal about this just after we enable it.

scaronni commented:
I've added the fallback service and made sure that the `nouveau.modeset=0` parameter is removed on Fedora 25+.

https://github.com/negativo17/nvidia-driver/commit/6b3bf3299bcc0069257590a72eb107b3acd8e7dd
https://github.com/negativo17/nvidia-driver/commit/df32483012fea9962c6ed41c61c377121c0b222b

I'm going to make some test (hopefully this evening) and write a blog post/change the docs to announce that the packages no longer upgrade ELRepo, RPMFusion and Nvidia packages. I have also a couple of pending items to explain so this is a good moment.

After that, so I guess this weekend or next week, I'll push the ones with Conflicts in place. I need to also make the same changes to the CUDA and NVENC packages to avoid errors in solving dependencies when you have/want the other packages.

jwrdegoede commented:
Hi,

On 09-05-17 10:17, Simone Caronni wrote:
> *@scaronni* commented on this pull request.
> 
> ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> 
> In nvidia-driver.spec <https://github.com/negativo17/nvidia-driver/pull/20#discussion_r115431648>:
> 
>> @@ -4,15 +4,15 @@
>   %if 0%{?rhel} == 6
>   # RHEL 6 does not have _udevrulesdir defined
>   %global _udevrulesdir   %{_prefix}/lib/udev/rules.d/
> -%global _dracutopts     nouveau.modeset=0 rdblacklist=nouveau
> -%global _dracutopts_rm  nomodeset vga=normal
> +%global _dracutopts     rdblacklist=nouveau
> +%global _dracutopts_rm  nomodeset vga=normal nouveau.modeset=0
> 
> Further down in the SPEC file I just check for the whole string of options, and the other two have been there for quite a while, so I guess all users will not have those and the |nouveau.modeset=0| parameter would not get removed.
> 
> I guess I need to remove the old Dracut options and just leave that, or iterate on each of one in the SPEC file.

Ah good point, I would prefer to iterate and remove then one-by-one,
to make sure we get things right on upgrade too and not just on a
clean install.

Regards,

Hans

jwrdegoede commented:
Hi,

On 09-05-17 10:20, Simone Caronni wrote:
> Looks fine for me, as written in the email I would have actually preferred to have the kernel locked into place until all modules are available.

We also need the fallback for people who install the nvidia driver on
systems where it cannot work because e.g. the nvidia GPU is too old and
since we are doing dkms/akmods we should basically never not have the
module available. I know that major kernel version bumps may cause
the build too fail, but we can see those coming long before hand
and normally we should have fixed dkms/akmods packages available
in time for things to just work when the new kernel hits the updates
repository.

> Someone who installs the Nvidia drivers would definitely give priority to the drivers being available than updating the kernel. I think users will be pretty vocal about this just after we enable it.

See above, the fallback is intended for when things are seriously
broken or simply just cannot work. Almost all kernel updates or
in the .z part of the x.y.z version number and there akmods/dkms
should just work.

Regards,

Hans

jwrdegoede commented:
Hi,

On 11-05-17 16:47, Simone Caronni wrote:
> I've added the fallback service and made sure that the |nouveau.modeset=0| parameter is removed on Fedora 25+.

Thank you.

> negativo17/nvidia-driver@6b3bf32 <https://github.com/negativo17/nvidia-driver/commit/6b3bf3299bcc0069257590a72eb107b3acd8e7dd>

In this one you also removed "blacklist nouveau" from nvidia.conf,
we actually do still need that, since nvidia.ko is always loaded
post initrd, otherwise udev will modprobe both and which one we
get is unknown, so that needs to be fixed.

The fallback service will work regardless because an explicit
modprobe (rather then one from udev) will ignore the blacklist.

> negativo17/nvidia-driver@df32483 <https://github.com/negativo17/nvidia-driver/commit/df32483012fea9962c6ed41c61c377121c0b222b>

This one looks good.

> 
> I'm going to make some test (hopefully this evening) and write a blog post/change the docs to announce that the packages no longer upgrade ELRepo, RPMFusion and Nvidia packages. I have also a couple of pending items to explain so this is a good moment.
> 
> After that, so I guess this weekend or next week, I'll push the ones with Conflicts in place. I need to also make the same changes to the CUDA and NVENC packages to avoid errors in solving dependencies when you have/want the other packages.

Ok, as always thank you for your work on this.

Regards,

Hams

jwrdegoede commented:
Hmm, my latest push was meant for a new pull-req really, since 2 of the 3 commits from this pull-req have been merged and the other one is going to get a different fix. So lets close this pull-req.
